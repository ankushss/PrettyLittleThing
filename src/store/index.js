import {configureStore} from '@reduxjs/toolkit';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import AsyncStorage from '@react-native-async-storage/async-storage';

import {dressApi} from './dress';
import {cartReducer} from './cart';

const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
};

const persistedCardReducer = persistReducer(persistConfig, cartReducer);

export const store = configureStore({
  reducer: {
    [dressApi.reducerPath]: dressApi.reducer,
    cart: persistedCardReducer,
  },
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }).concat(dressApi.middleware),
});

export const persistor = persistStore(store);
