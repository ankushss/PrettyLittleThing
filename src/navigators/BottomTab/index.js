import React from 'react';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';

import {Home} from '../../screens/Home';
import {Cart} from '../../screens/Cart';
import {Image, StyleSheet} from 'react-native';
import {useSelector} from 'react-redux';

import {icons} from '../../assets/icons';
import {getCartItemsCount} from '../../store/cart';

const Tab = createBottomTabNavigator();

export const BottomTab = () => {
  const cartCount = useSelector(getCartItemsCount);
  return (
    <Tab.Navigator
      screenOptions={{
        tabBarActiveTintColor: '#00aa5b',
      }}>
      <Tab.Screen
        name="Home"
        component={Home}
        options={{
          tabBarIcon: ({focused, color, size}) => (
            <Image
              source={icons.home}
              style={styles.icon(focused, color, size)}
            />
          ),
        }}
      />
      <Tab.Screen
        name="Cart"
        component={Cart}
        options={{
          tabBarIcon: ({focused, color, size}) => (
            <Image
              source={icons.cart}
              style={styles.icon(focused, color, size)}
            />
          ),
          tabBarBadge: cartCount > 0 ? cartCount : null,
        }}
      />
    </Tab.Navigator>
  );
};

const styles = StyleSheet.create({
  icon: (focused, color, size) => ({
    width: size,
    height: size,
    tintColor: color,
  }),
});
