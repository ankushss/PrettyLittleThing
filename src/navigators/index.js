import React from 'react';
import {NavigationContainer} from '@react-navigation/native';

import {BottomTab} from './BottomTab';

export const Navigator = () => {
  return (
    <NavigationContainer>
      <BottomTab />
    </NavigationContainer>
  );
};
