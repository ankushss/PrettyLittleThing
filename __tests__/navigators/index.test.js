import React from 'react';
import {create} from 'react-test-renderer';
import {Navigator} from '../../src/navigators';

describe('Navigator', () => {
  it('matches snapshot', () => {
    const component = create(<Navigator />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
