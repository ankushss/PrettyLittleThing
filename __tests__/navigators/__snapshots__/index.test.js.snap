// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigator matches snapshot 1`] = `
<NavigationContainer>
  <Navigator
    screenOptions={
      Object {
        "tabBarActiveTintColor": "#00aa5b",
      }
    }
  >
    <Screen
      component={[Function]}
      name="Home"
      options={
        Object {
          "tabBarIcon": [Function],
        }
      }
    />
    <Screen
      component={[Function]}
      name="Cart"
      options={
        Object {
          "tabBarBadge": null,
          "tabBarIcon": [Function],
        }
      }
    />
  </Navigator>
</NavigationContainer>
`;
